src/flp.rs

```diff
@@ -543,6 +548,13 @@ pub trait Type: Sized + Eq + Clone + Debug {
 
         Ok(())
     }
+
+    #[cfg(feature = "experimental")]
+    /// Optionally add noise to the aggregate share.
+    /// Post-condition: The size of `_aggregate_share` has not changed.
+    fn add_noise(&self, _aggregate_share: &mut Vec<Self::Field>) -> Result<(), FlpError> {
```

Comment:
Yes, it could, of course. I have floated the [idea](https://github.com/divviup/libprio-rs/pull/578#discussion_r1228082901) to maybe change this function to be
```
fn generate_noise(&self) -> Result<Vec<F>, FlpError>
```
instead. This would obviously require allocating a new vector, so from an efficiency standpoint the current variant (updated to use slices) would be preferable, though slightly less general.

There is also the option to go ahead with the `add_noise` variant (updated to use slices), and, in order to generate a vector containing only noise, apply it to a "zero-initialized" share. The value for "zero" would depend on the Prio3 type in question, and I assume would require an additional function in the Prio3 trait (which wouldn't need to be added now, as our usecase does not require pure noise shares).